init commit

Chengqian-Zhang opened the pull request.

wanghan-iapcm commented on tests/test_prep_run_vasp.py:
```diff
@@ -0,0 +1,212 @@
+    def setUp(self):
+        warnings.simplefilter('ignore', ResourceWarning)
+        self.ntasks = 3
+        self.confs = [Path(Path('confs')/'data.000'),Path(Path('confs')/'data.001')]
```
Do you really needs the conf files for the testing purpose?

Chengqian-Zhang replied:
I have deleted the conf files.

wanghan-iapcm reviewed (changes requested):
- do we need examples on prep_vasp and run_vasp? a user is not supposed to use the steps at all. 
- please remove all the data files from the examples. Only provide the minimal set of files that are needed to run the examples.
- the examples are associated to the website of workflows.deepmodeling.com, which is not recommended. 
- python installation and distribution is missing
- the name convention of the modules and files are not good.

Chengqian-Zhang commented:
> * do we need examples on prep_vasp and run_vasp? a user is not supposed to use the steps at all.
I have deleted the examples on prep_vasp and run_vasp.
> * please remove all the data files from the examples. Only provide the minimal set of files that are needed to run the examples.
I have updated the example of preprun_vasp. But It still has a "POTCAR" file. I don't know if this is appropriate.
> * the examples are associated to the website of workflows.deepmodeling.com, which is not recommended.
I have deleted the related infomation about the website of workflows.deepmodeling.com.
> * python installation and distribution is missing
I don't quite understand what "python installation and distribution" refers to.
> * the name convention of the modules and files are not good.
I have changed the structure and name of the directory. Please check again.

Chengqian-Zhang commented on tests/test_prep_run_vasp.py:
```diff
@@ -0,0 +1,212 @@
+
+class TestMockedRunVasp(unittest.TestCase):
+    def setUp(self):
+        warnings.simplefilter('ignore', ResourceWarning)
```
I have deleted the sentence.

Chengqian-Zhang commented on PrepRunFp/PrepRunFp.py:
```diff
@@ -0,0 +1,266 @@
+            "inputs" : InputParameter(),
+            "optional_input" : InputParameter(),
+            "type_map" : InputParameter(),
+            "backward_list" : InputParameter(),
```
I change them to:
"optional_input" : InputParameter(type=dict , value={})
"inputs","type_map" and "backward_list" are mandatory.

Chengqian-Zhang commented on PrepRunFp/PrepRunFp.py:
```diff
@@ -0,0 +1,266 @@
+        self._input_parameters = {
+            "config" : InputParameter(),
+            "inputs" : InputParameter(),
+            "optional_input" : InputParameter(),
```
I change them to:
"config" : InputParameter(type=dict , value={})
"optional_input" : InputParameter(type=dict , value={})
"inputs" is mandatory.
Optional parameters have been placed after compulsory parameters.

Chengqian-Zhang commented on PrepRunFp/PrepRunFp.py:
```diff
@@ -0,0 +1,266 @@
+        }
+        self._input_artifacts = {
+            "confs" : InputArtifact(),
+            "optional_artifact" : InputArtifact(), 
```
I change them to:
"confs" : InputArtifact(),
"optional_artifact" : InputArtifact(optional=True),

Chengqian-Zhang commented on tests/test_prep_run_vasp.py:
```diff
@@ -0,0 +1,212 @@
+                "config" : {},
+                "optional_artifact" : {},
+                "backward_list" : ['POSCAR','POTCAR'],
+                "optional_input" : {},
```
I have added tests for the case when the optional parameters/artifacts are not provided.Please check.

njzjz commented on LICENSE:
```diff
@@ -0,0 +1,21 @@
+MIT License
```
MIT is not compatible with LGPL

Chengqian-Zhang replied:
I have changed LICENSE to LGPL.

njzjz commented on fpop/run_fp.py:
```diff
@@ -0,0 +1,197 @@
+@contextmanager
+def set_directory(path: Path):
+    '''Sets the current working path within the context.
+    Parameters
```
There should be a blank line be fore each section

Chengqian-Zhang replied:
I have added the blank lines.

njzjz commented on setup.py:
```diff
@@ -0,0 +1,31 @@
+    long_description = fh.read()
+
+setuptools.setup(
+    name="fpoptest",
```
Do we want to use this name?

Chengqian-Zhang replied:
I change the name to fpop.But I have not submitted to pypi.

wanghan-iapcm commented on fpop/preprun_fp.py:
```diff
@@ -0,0 +1,187 @@
+        run_op : OP,
+        prep_image : str,
+        run_image : str,
+        prep_config : Optional[dict] = None,
+        run_config : Optional[dict] = None,
```
It seems that the input config should not be one simple dict. 
The parameters passed to template, slice and the step may be provided separately.

wanghan-iapcm commented on fpop/preprun_fp.py:
```diff
@@ -0,0 +1,187 @@
+            "type_map" : InputParameter(),
+            "backward_list" : InputParameter(),
+            "config" : InputParameter(type=dict , value={}),
+            "optional_input" : InputParameter(type=dict , value={}),
```
Please provide doc explaining what are the acceptable keys in the optional input.

wanghan-iapcm commented on fpop/preprun_fp.py:
```diff
@@ -0,0 +1,187 @@
+        }
+        self._input_artifacts = {
+            "confs" : InputArtifact(),
+            "optional_artifact" : InputArtifact(optional=True), 
```
Please provide doc explaining the `optional_artifact`

wanghan-iapcm commented on fpop/preprun_fp.py:
```diff
@@ -0,0 +1,187 @@
+            "inputs" : InputParameter(),
+            "type_map" : InputParameter(),
+            "backward_list" : InputParameter(),
+            "config" : InputParameter(type=dict , value={}),
```
we need one config for prep and one for run.

The name of the `config` may be confused with the `config` of `dflow.Step`. We may rename it as `image_config`

Chengqian-Zhang replied:
I split this "config" to "prep_image_config" and "run_image_config".For example, "command" is a keyward in "run_image_config"

Chengqian-Zhang commented on README.md:
```diff
@@ -0,0 +1,2 @@
+# FP_OPs
```
I have updated README.

AnguseZhang commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,38 @@
+        export DFLOW_S3_REPO_KEY=${{ secrets.DFLOW_S3_REPO_KEY }}
+        export DFLOW_S3_STORAGE_CLIENT=${{ secrets.DFLOW_S3_STORAGE_CLIENT }}
+        export BOHRIUM_USERNAME=${{ secrets.BOHRIUM_USERNAME }}
+        export BOHRIUM_PASSWORD=${{ secrets.BOHRIUM_PASSWORD }}
```
Delete these if you do not need.

Chengqian-Zhang replied:
The bug of dflow is fixed.

AnguseZhang commented on README.md:
```diff
@@ -0,0 +1,2 @@
+# fpop
+`fp` stands for first principle calculation and `op` stands for operators. The abbreviation `fpop` stands for operators related to first principle calculation.
```
first-principles

AnguseZhang replied:
Suggest add some basic background here, such as based on `dflow` (attached link). Furnish the words.

AnguseZhang commented on fpop/prep_fp.py:
```diff
@@ -0,0 +1,178 @@
+        os.chdir(path)
+        yield
+    finally:
+        os.chdir(cwd)
```
Use the tools `set_directory` in `dflow`

Chengqian-Zhang replied:
I have done it.

AnguseZhang commented on setup.py:
```diff
@@ -0,0 +1,31 @@
+    description="operators about first principle caculations",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    url="https://github.com/Chengqian-Zhang/FP_OPs",
```
Update url and necessary information

Chengqian-Zhang replied:
I have updated url.

AnguseZhang commented:
@Chengqian-Zhang Please solve the minor changes I requested. Then I will merge this PR.

Chengqian-Zhang commented on README.md:
```diff
@@ -0,0 +1,2 @@
+# fpop
+`fp` stands for first-principles calculation and `op` stands for operators. The abbreviation `fpop` stands for operators related to first-principles calculation.This project is base on [dflow](https://github.com/deepmodeling/dflow) which is a Python framework for constructing scientific computing workflows (e.g. concurrent learning workflows) employing Argo Workflows as the workflow engine. 
```
I have fixed it.
